R/eml.R

```diff
@@ -521,60 +544,84 @@ eml_individual_name <- function(given_names=NULL, sur_name) {
   indiv_name
 }
 
-
 #' Create an eml-project section.
 #'
-#' Note: This is super-limited right now.
+#' Note - studyAreaDescription, designDescription, and relatedProject are not fully fleshed out. Need to pass these objects in directly if you want to use them.
 #'
-#' @param title (character) Title of the project.
-#' @param awards (character) One or more awards for the project.
-#' @param first (character) First name of the person with role `role`.
-#' @param last (character) Last name of the person with role `role`.
-#' @param organizations (character) Optional. One or more organization strings.
-#' @param role (character) Optional. Specify an alternate role.
+#' @param title (character) Title of the project (Required).
+#' @param personnelList (list of eml_personnel) Personnel involved with the project (role:originator required).
+#' @param abstract (character) Project abstract.
+#' @param funding (character) Funding sources for the project such as grant and contract numbers.
+#' @param studyAreaDescription (studyAreaDescription) 
+#' @param designDescription (designDescription) 
+#' @param relatedProject (project) 
 #'
 #' @return (project) The new project section.
 #' @export
 #'
 #' @examples
-#' eml_project("Some title", "51231", "Some", "User")
-eml_project <- function(title, awards, first, last, organizations = NULL, role = "originator") {
-  stopifnot(all(sapply(c(title, awards, first, last), is.character)),
-            all(lengths(c(title, awards, first, last)) > 0))
+#' eml_project("Some title", list(personnel1, personnel2), "Abstract", "#1 Best Scientist Award")
+eml_project <- function(title, personnelList, abstract = NULL, funding = NULL, studyAreaDescription = NULL, designDescription = NULL, relatedProject = NULL) {
+  
+  stopifnot(is.character(title),
+            nchar(title) > 0)
+  stopifnot(length(personnelList) > 0)
 
-  # project
+  # Project
   project <- new("project")
 
-  # title
-  title_ele <- new("title")
-  title_ele@.Data <- title
-  project@title <- new("ListOftitle", list(title_ele))
-
-  # personnel
-  personnel <- new("personnel")
+  # Title
+  project@title <- c(new("title", .Data = title))
 
-  # individualName
-  personnel@individualName <- new("ListOfindividualName", list(eml_individual_name(first, last)))
-
-  # organizationName
-  if (!is.null(organizations)) {
-    organizations <- lapply(organizations, function(org) { o <- new("organizationName"); o@.Data <- org; o } )
-    personnel@organizationName <- new("ListOforganizationName", organizations)
+  # Personnel
+  if(!all(sapply(personnelList, function(x) { is(x, "personnel") }))) {
+    stop(call. = FALSE,
+         "All personnel in the list must be of type 'personnel'")
+  }
+  
+  if(!all(sapply(personnelList, function(x) { length(x@role) > 0 }))) {
```

Comment:
Do you think this check might be better if put in the `eml_personnel` helper instead since it's more of a concern of that object, rather than the project?